### PR TITLE
Make category a single-value field shared by all media types

### LINKED ISSUE #357: Category must be a single value string
## Description

The API returns `category` as a list.

## Expectation

Category should be a string, not an array. 

## Additional context

Also `category` should be defined as a column on the base media model, ensuring consistency between the image and audio models. The ingestion server will need to be updated to ingest the column for images like it does for audio.

## Resolution

- [x] 🙋 I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by dhruvkb**
## Fixes

Fixes #357 by @dhruvkb 

## Description

This PR
- moves the `category` field from `Audio` to `AbstractMedia` to use it on all media models
- updates categories in the sample audio data from arrays to string fields

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by zackkrida** [comment]
Looks great so far

**Comment by dhruvkb** [comment]
The main issue with this lies in the underlying data (https://github.com/WordPress/openverse/issues/1683). The sample data does not contain the `category` column so it does not get copied into the API database and so the detail view (at `/v1/images/<id>/`) shows `category` as `null`.

Interestingly the `categorize.py` file fills the correct info into ElasticSearch so searching by category works, and the search results view shows categories too. Just the detail view, which fetches the API database info has this bug.

File: ingestion_server/ingestion_server/elasticsearch_models.py [hunk 1]
```diff
@@ -241,7 +241,7 @@ class Index:
     def database_row_to_elasticsearch_doc(row, schema):
         source = row[schema["source"]]
         extension = Image.get_extension(row[schema["url"]])
-        categories = get_categories(extension, source)
+        fallback_category = get_category(extension, source)
```
Comment (zackkrida) [inline ingestion_server/ingestion_server/elasticsearch_models.py:244]: Great approach here!

**Review by zackkrida — APPROVED** [review]
Nicely done. The backwards compatibility and fallback for image categories is a very nice detail.

File: openverse_api/catalog/api/models/image.py [hunk 2]
```diff
@@ -29,7 +29,11 @@ class Meta:
 
 
 class Image(ImageFileMixin, AbstractMedia):
-    pass
+    """
+    Inherited fields
+    ================
+    category: eg. photograph, digitised artwork & illustration
```
Comment (AetherUnbound) [inline openverse_api/catalog/api/models/image.py:35]: Should we include all inherited fields here?

Comment (dhruvkb) [reply openverse_api/catalog/api/models/image.py:35]: We should, but so far only the `category` field has a documentation-worthy note.

**Review by obulat — APPROVED** [review]

**Review by krysal — CHANGES_REQUESTED** [review]
Looks great but `test_image_success_examples` failed when I ran it (weird that it did pass in the ci workflow). It's just a minor change adding the `category` field to the example image JSON.

**Review by krysal — APPROVED** [review]
Nice! Everything looks good to me  ⭐️

**Pull request merged**